lib/engine/game/g_18_ts/game.rb

```diff
@@ -0,0 +1,1166 @@
+            Engine::Step::HomeToken,
+            G18TS::Step::TrackAndToken,
+            G18TS::Step::Route,
+            G18TS::Step::Dividend,
+            Engine::Step::DiscardTrain,
+            G18TS::Step::BuyTrain,
+          ], round_num: round_num)
+        end
+
+        # Minors need to have a tile with cities and paths before they can lay
+        # their home token
+        def can_place_home_token?(entity)
+          return true unless minor?(entity)
+
+          home_hex = hex_by_id(entity.coordinates)
+          raise GameError, "can't find home hex for #{entity.name}" unless home_hex
+
+          !home_hex.tile.paths.empty? && !home_hex.tile.cities.empty?
+        end
+
+        def place_home_token(corporation)
+          return if corporation.tokens.first&.used
+          return unless can_place_home_token?(corporation)
+
+          unless corporation.coordinates.is_a?(Array)
+            super
+            @graph.clear
+            return
+          end
+
+          corporation.coordinates.each do |coord|
+            hex = hex_by_id(coord)
+            next unless hex
+
+            tile = hex&.tile
+            next unless tile
+
+            cities = tile.cities
+            city = cities.find { |c| c.reserved_by?(corporation) } || cities.first
+            next unless city
+
+            token = corporation.find_token_by_type
+
+            @log << "#{corporation.name} places a token on #{hex.name}"
+            city.place_token(corporation, token)
+          end
+          @graph.clear
+        end
+
+        # Formation isn't flotation for minors
+        def formed?(corp)
+          @formed.include?(corp)
+        end
+
+        # For minors: not flotation, but when minor can purchase its required train
+        def check_formation(corp)
+          return if formed?(corp)
+
+          if major?(corp) && corp.floated?
+            @formed << corp
+            @log << "#{corp.name} forms"
+          elsif minor?(corp) && corp.cash >= minor_required_train(corp).price
+            @formed << corp
+            @log << "Minor #{corp.name} forms"
+
+            # buy required train (no phase-change side-effects)
+            r_train = minor_get_train(corp)
+            @minor_trains.delete(r_train)
+            corp.trains << r_train
+            r_train.owner = corp
+            corp.spend(r_train.price, @bank)
+            @log << "Minor #{corp.name} spends #{format_currency(r_train.price)} for required train (#{r_train.name})"
+          end
+        end
+
+        # -1 if a has a higher par price than b
+        # 1 if a has a lower par price than b
+        # if they are the same, then use the order of formation (generally flotation)
+        def par_compare(a, b)
+          if a.par_price.price > b.par_price.price
+            -1
+          elsif a.par_price.price < b.par_price.price
+            1
+          else
+            @formed.find_index(a) < @formed.find_index(b) ? -1 : 1
```

Comment:
You can use the spaceship operator here:
```suggestion
            @formed.find_index(a) <=> @formed.find_index(b)
```